src/dshould/stringcmp.d

```diff
@@ -0,0 +1,441 @@
+module dshould.stringcmp;
+
+import std.algorithm : EditOp, map;
+import std.format : format;
+import std.range;
+import std.typecons;
+import dshould.ShouldType;
+import dshould.basic;
+
+void equal(Should, T)(Should should, T expected, string file = __FILE__, size_t line = __LINE__)
+if (isInstanceOf!(ShouldType, Should) && is(T == string))
+{
+    should.allowOnlyWordsBefore!([], "equal (string)");
+
+    should.addWord!"equal".stringCmp(expected, file, line);
+}
+
+void stringCmp(Should, T)(Should should, T expected, string file, size_t line)
+if (isInstanceOf!(ShouldType, Should))
+{
+    import std.algorithm : canFind;
+
+    should.allowOnlyWordsBefore!(["equal"], "");
+
+    should.terminateChain;
+
+    auto got = should.got();
+
+    if (got != expected)
+    {
+        string original;
+        string diff;
+
+        if (got.canFind("\n"))
+        {
+            auto diffPair = multiLineDiff(expected.split("\n"), got.split("\n"));
+
+            original = diffPair.original.join("\n") ~ "\n";
+            diff = "\n" ~ diffPair.target.join("\n");
+        }
+        else
+        {
+            auto diffPair = oneLineDiff(expected, got);
+
+            original = diffPair.original;
+            diff = diffPair.target;
+        }
+
+        throw new FluentException(
+            "test failed",
+            format!`: expected '%s', but got '%s'`(original, diff),
+            file, line
+        );
+    }
+}
+
+@("collates successive replacements")
+unittest
+{
+    const expectedOriginal = "Hello W" ~ RED_CODE ~ "or" ~ CLEAR_CODE ~ "ld";
+    const expectedTarget = "Hello W" ~ GREEN_CODE ~ "ey" ~ CLEAR_CODE ~ "ld";
+    const diff = "Hello World".oneLineDiff("Hello Weyld");
+
+    (diff.original ~ diff.target).should.equal(expectedOriginal ~ expectedTarget);
```

Comment:
concatenation is a hack; would `tuple` comparison work?

Reply:
~ is used because when the diff algorithm fails, you are almost always interested in both halves - and you usually also want visible color coding, which breaks with Tuple.toString.